lunatrace/bsl/backend/src/graphql-yoga/resolvers/create-pull-request-for-vulnerability.ts

```diff
@@ -0,0 +1,142 @@
+export const createPullRequestForVulnerabilityResolver: CreatePullRequestForVulnerabilityType = async (
+  parent,
+  args,
+  ctx,
+  _info
+) => {
+  try {
+    if (!isAuthenticated(ctx)) {
+      log.warn('No parsed JWT claims with a user ID on route that required authorization, throwing a graphql error');
+      throw new GraphQLYogaError('Unauthorized');
+    }
+
+    const vulnerabilityId = args.vulnerability_id;
+    if (!vulnerabilityId) {
+      throw new GraphQLYogaError('No vulnerability id provided');
+    }
+
+    const projectId = args.project_id;
+    if (!projectId) {
+      throw new GraphQLYogaError('No project id provided');
+    }
+
+    const oldPackageSlug = args.old_package_slug;
+    if (!oldPackageSlug) {
+      throw new GraphQLYogaError('No old package slug provided');
+    }
+
+    const newPackageSlug = args.new_package_slug;
+    if (!newPackageSlug) {
+      throw new GraphQLYogaError('No new package slug provided');
+    }
+
+    const packageManifestPath = args.package_manifest_path;
+    if (!packageManifestPath) {
+      throw new GraphQLYogaError('No package manifest path provided');
+    }
+
+    // Strips out the actual package.json/yarn.lock/package-lock.json from the path since we just need the folder.
+    const normalizedPackageManifestPath = packageManifestPath
+      .replace(/package\.json$/, '')
+      .replace(/yarn\.lock$/, '')
+      .replace(/package-lock\.json$/, '');
+
+    await checkProjectIsAuthorizedOrThrow(projectId, ctx);
+
+    const installationIdResult = await hasura.GetGitHubInstallationIdFromProjectId({
+      project_id: projectId,
+    });
+
+    if (installationIdResult.projects.length === 0) {
+      throw new GraphQLYogaError('No project found when fetching installation id');
+    }
+
+    const installationId = installationIdResult.projects[0].organization?.installation_id;
+
+    if (!installationId) {
+      throw new GraphQLYogaError('No installation id found for project');
+    }
+
+    const githubRepo = installationIdResult.projects[0].github_repository?.git_url;
+
+    if (!githubRepo) {
+      throw new GraphQLYogaError('No github repo found for project');
+    }
+
+    const { owner, repo } = splitGitHubRepoPath(githubRepo);
+
+    const installationAccessTokenRes = await getInstallationAccessToken(installationId);
+    if (installationAccessTokenRes.error) {
+      log.error('Failed to fetch installation access token', { installationAccessTokenRes });
+      throw new Error('Error fetching installation access token');
+    }
+
+    const installationAccessToken = installationAccessTokenRes.res;
+
+    const octokit = new PullRequestOctokit({
+      auth: installationAccessToken,
+    });
+
+    const pullRequestResult = await replacePackageAndFileGitHubPullRequest(
+      octokit,
+      owner,
+      repo,
+      normalizedPackageManifestPath,
+      // TODO: Make this less hacky and brittle
+      packageManifestPath.endsWith('package-lock.json') ? 'npm' : 'yarn',
```

Comment:
This is probably not a safe check. I think we should expand this into a function that throws if it doesnt look npm-y or yarn-y.

Pnpm exists, and so on and so forth. Bun too! haha